sql/expression/function/lower_upper.go

```diff
@@ -0,0 +1,120 @@
+package function
+
+import (
+	"fmt"
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+	"gopkg.in/src-d/go-mysql-server.v0/sql/expression"
+	"strings"
+)
+
+// Lower is a function that returns the lowercase of the text provided.
+type Lower struct {
+	expression.UnaryExpression
+}
+
+// NewLower creates a new Lower expression.
+func NewLower(e sql.Expression) sql.Expression {
+	return &Lower{expression.UnaryExpression{Child: e}}
+}
+
+// Eval implements the Expression interface.
+func (l *Lower) Eval(
+	ctx *sql.Context,
+	row sql.Row,
+) (interface{}, error) {
+	v, err := l.Child.Eval(ctx, row)
+	if err != nil {
+		return nil, err
+	}
+
+	if v == nil {
+		return nil, nil
+	}
+
+	if !sql.IsText(l.Child.Type()) {
```

Comment:
According to MySQL behaviour, the argument is converted to text, then lowered. So you can just do:

```go
v, err = sql.Text.Convert(v)
if err != nil {
        return nil, err
}

return strings.ToLower(v.(string)), nil
```